src/Dotnet.Script.Core/ScriptPublisher.cs

```diff
@@ -63,20 +63,21 @@ public ScriptPublisher(LogFactory logFactory, ScriptEmitter scriptEmitter)
             }
         }
 
-        public void CreateExecutable<TReturn, THost>(ScriptContext context, LogFactory logFactory, string runtimeIdentifier)
+        public void CreateExecutable<TReturn, THost>(ScriptContext context, LogFactory logFactory, string runtimeIdentifier, string executeFileName = null)
         {
             if (runtimeIdentifier == null)
             {
                 throw new ArgumentNullException(nameof(runtimeIdentifier));
             }
 
             const string AssemblyName = "scriptAssembly";
+            executeFileName = executeFileName ?? Path.GetFileNameWithoutExtension(context.FilePath);
 
-            var tempProjectPath = ScriptProjectProvider.GetPathToProjectFile(Path.GetDirectoryName(context.FilePath), ScriptEnvironment.Default.TargetFramework);
+            var tempProjectPath = ScriptProjectProvider.GetPathToProjectFile(Path.GetDirectoryName(context.FilePath), ScriptEnvironment.Default.TargetFramework, executeFileName);
             var tempProjectDirectory = Path.GetDirectoryName(tempProjectPath);
 
             var scriptAssemblyPath = CreateScriptAssembly<TReturn, THost>(context, tempProjectDirectory, AssemblyName);
-            var projectFile = new ProjectFile(File.ReadAllText(tempProjectPath));
+            var projectFile = new ProjectFile();
```

Comment:
What's the purpose of reading temporary project file here? 
Seems that there has been a script.csproj inside the temporary path, but I don't know when and where these files are written.